### PR TITLE
cli: new README.md to explain general UX guidelines

### PR DESCRIPTION
**Pull request by knz**
cc @thtruo @mikeCRL @taroface 

The goal of this PR is to make it possible for future contributors to augment the CLI while preserving consistent UX.

Release note: None

### CODE REVIEW

File: pkg/cli/README.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# CLI utility design guidelines
+
+## Guidelines for operator-level utilities
+
+- Be concise in the common case, verbose with errors
```
Comment (otan) [inline pkg/cli/README.md:5]: super nit: full stop end some bullet points but not others

File: pkg/cli/README.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    In this mode, there should *not* be guidance printed out to help a
+    human.
+
+    Also, the automated input mode is part of the public API of CockroachDB.
+    Beware of documenting changes in release notes; use the deprecation cycle
+    to remove key input aspects; ensure that users can preserve specific
+    outputs from previous version when upgrading, possibly via opt-in flags.
+
+- Beware: the two modes above are orthogonal to each other. We know
```
Comment (otan) [inline pkg/cli/README.md:59]: super nit: feels like there should be a "the" between "know" and "usage"

**Review by otan — APPROVED** [review]

File: pkg/cli/README.md (hunk 1 above)
Comment (knz) [reply pkg/cli/README.md:5]: Fixed.

File: pkg/cli/README.md (hunk 2 above)
Comment (knz) [reply pkg/cli/README.md:59]: done.

**Comment by knz** [comment]
bors r=otan

**Comment by knz** [comment]
unrelated flake

bors r=otan

**Pull request merged**